src/axom/core/tests/core_flatmap.hpp

```diff
@@ -70,10 +71,12 @@ class core_flatmap : public ::testing::Test
   using KeyType = typename FlatMapType::key_type;
   using ValueType = typename FlatMapType::mapped_type;
 
-#if defined(AXOM_USE_HIP)
+#if defined(AXOM_RUNTIME_POLICY_USE_HIP)
   using DeviceExec = axom::HIP_EXEC<256>;
-#elif defined(AXOM_USE_CUDA)
+#elif defined(AXOM_RUNTIME_POLICY_USE_CUDA)
   using DeviceExec = axom::CUDA_EXEC<256>;
```

Comment:
What's the difference here between `AXOM_USE_HIP/CUDA` and `AXOM_RUNTIME_POLICY_USE_HIP/CUDA`?

Reply:
As an example, ``AXOM_RUNTIME_POLICY_USE_CUDA`` means that ``AXOM_USE_RAJA``, ``AXOM_USE_UMPIRE``, and ``AXOM_USE_CUDA`` are all defined. Those are the conditions under which ``axom::CUDA_EXEC`` is defined. The situation is similar for HIP. There have been several places in the code where some but not all macros were checked before using ``axom::CUDA_EXEC``. This caused problems in configurations where CUDA is defined but RAJA is not. Using the ``AXOM_RUNTIME_POLICY_USE_CUDA`` (which arose later) was a safer/simpler way (IMHO) to access conditionally-compiled execution policies.

Reply:
Thanks for the explanation, that does sound like a much cleaner solution.

Although I wonder, what GPU functionality would a user even be able to use from us by compiling Axom with CUDA/HIP, but no RAJA or Umpire support?

Reply:
I do worry that we are trying to support too many configurations and trying to avoid requiring RAJA and Umpire is actually leading to more complicated code.

Reply:
There's pushback on the other side from users who want to use a subset of axom with fewer external dependencies. 
E.g. users who only want `sina` (which requires `conduit`, and optionally `hdf5`, but nothing else).

Reply:
Does sina actually use any of this functionality? I thought it was all CPU-only code.